azkaban-web-server/src/main/java/azkaban/webapp/servlet/ProjectManagerServlet.java

```diff
@@ -1728,6 +1835,9 @@ private void ajaxHandleUpload(final HttpServletRequest req, final HttpServletRes
           this.scheduler.schedule(project, user.getUserId());
         }
 
+        // reset locks for flows as needed
+        lockFlowsForProject(project, lockedFlows);
```

Comment:
Can a user get rid of the lock just by changing the flow name and re-upload?

Reply:
Yes, if the flow is renamed, or if the flow is deleted and added back to the project, the lock would be removed. The lock is tied to the flow name.